.grit/patterns/mui4_to_mui5.md

```diff
@@ -0,0 +1,27 @@
+## title: Convert Rename theme.palette.type In MUI v4 To MUI v5
+
+This pattern Rename theme.palette.type in MUI v4 to MUI v5
+
+tags: #react, #migration, #complex
+
+```grit
+engine marzano(0.1)
+language js
+
+// Most of the logic for this pattern is in mui4_to_mui5.grit
+// https://github.com/getgrit/js/blob/main/.grit/patterns/mui4_to_mui5.grit
+
+`createTheme({ palette: { type: $arg } })` => `createTheme({ palette: { mode: $arg } })` where {
```

Comment:
This pattern is not sufficiently general. Ex. this won't match a theme that has anything except a palette.

Here's how you would generalize it more.
```
`createTheme($theme)` where {
  $theme <: contains `palette: { $palette }`,
  $palette <: contains `type: $arg` => `type: $arg`
}
```

Reply:
This has been updated @morgante